functions/public/Update-omsAgent.ps1

```diff
@@ -0,0 +1,179 @@
+#Requires -Version 5.0
+
+function Update-OmsAgent
+{
+	<#
+		.Synopsis
+			Update the OMS agent on remote computers.
+		.DESCRIPTION
+			Either downloads the installer from a URL or copies the installer via the powershell session. Can detected if a previous version is installed and skip if so. If allready installed WorkSpaceId and WorkSpaceKey added to previous install. Doesn't detect invalid workspace IDs or Keys.
+		.EXAMPLE
+			Update-OmsAgent -sourcePath 'c:\MMASetup-AMD64.exe'  -Verbose
+		.EXAMPLE
+			Update-OmsAgent -computerName <computerName>  -Verbose
+		.NOTES
+			Written by ben taylor and Monty Harris
+			Version 1.0, 25.07.2017
+	#>
+	[CmdletBinding(SupportsShouldProcess = $True, ConfirmImpact = 'Low', DefaultParameterSetName='downloadOMS')]
+	[OutputType([String])]
+	Param (
+		[Parameter(Mandatory = $false, Position = 0, ValueFromPipeline=$True, valuefrompipelinebypropertyname=$true)]
+		[ValidateNotNullOrEmpty()]
+		[Alias('IPAddress', 'Name')]
+		[string[]]
+		$computerName = $env:COMPUTERNAME,
+
+		[Parameter(Mandatory=$true, ParameterSetName='downloadOms')]
+		[ValidateNotNullOrEmpty()]
+		[string]
+		$downloadURL = 'http://download.microsoft.com/download/0/C/0/0C072D6E-F418-4AD4-BCB2-A362624F400A/MMASetup-AMD64.exe',
```

Comment:
I am unsure about specifying a download link here as it may not be a current one. My feeling is to make the user specify there own DL link so they can make sure it is current, incase the default link is a old version. Thought?

Reply:
I agree however it is very difficult to find the download link at least in experience. I will test the link and see what version is being downloaded or see if there is an aka link out there